### PR TITLE
[RFC] Identify games that use procedural textures

### PR DESCRIPTION
**Pull request by bunnei**
I'm looking for some assistance finding games that use procedural textures for reverse engineering. Developers, please ignore this, I'm not looking for a review. Instead, I'd like for testers that have a lot of games (and/or time to test) to run this branch on their game collections to help me find games that use procedural textures.

If a game tries to use a procedural texture, you'll immediately see the following message box (only once per game):

image

**This will only work with with the Qt GUI.** Furthermore, any triangles that use this texture should appear bright magenta-colored in-game.

Any help is greatly appreciated. Furthermore, any screenshots of where the procedural textures occur (and/or how to reproduce the results) would be helpful. Thanks!

### CODE REVIEW

File: src/citra_qt/bootmanager.cpp [hunk 1]
```diff
@@ -44,8 +44,12 @@ void EmuThread::run() {
             if (!was_active)
                 emit DebugModeLeft();
 
-            Core::RunLoop();
-
+            try {
+                Core::RunLoop();
+            } catch (const char* str) {
```
Comment (Subv) [inline src/citra_qt/bootmanager.cpp]: I love it

**Comment by Subv** [comment]
LGTM, insta merge pls

File: src/citra_qt/bootmanager.cpp (hunk 1 above)
Comment (neobrain) [inline src/citra_qt/bootmanager.cpp]: @Subv See, **this** is why we need exceptions! :D

Comment (bunnei) [inline src/citra_qt/bootmanager.cpp]: I actually do really like exceptions as a way to abort emulation and shutdown safely, or to notify the user with a GUI box (as I am here..). So yeah, we should do this.

**Comment by bunnei** [comment]
NOTE: This will NOT work in conjunction with tfarley's OGL branches. Those will short circuit the rasterizer code with the procedural texture test exists. Please test with the software renderer in this branch.

**Comment by bunnei** [comment]
We didn't find any games, and I'm not really working on this right now. Will resume at another time.

**Comment by Subv** [comment]
#2213

**Comment by JayFoxRox** [comment]
More importantly, we have a label for that now: https://github.com/citra-emu/citra/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Agpu%3Aproctex%20

Wether #2213 is a false positive will have to be tested. I could imagine it being a bug in the CPU JIT or something which accidentely calculates a wrong texture unit. As op claimed that it worked fine before.

**Pull request closed without merging**